src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewRenderer.php

```diff
@@ -194,4 +198,43 @@ private function handle(Request $request)
             throw $e;
         }
     }
+
+    /**
+     * This creates a new portal information based on the given information. This is necessary because it is possible
+     * that a webspace defines a language, which is not used in any portal. For this case we have to define our own
+     * fake PortalInformation object.
+     *
+     * @param string $webspaceKey
+     * @param string $locale
+     *
+     * @return PortalInformation
+     */
+    private function createPortalInformation($object, $id, $webspaceKey, $locale)
+    {
+        $webspace = $this->webspaceManager->findWebspaceByKey($webspaceKey);
+
+        if (!$webspace) {
+            throw new WebspaceNotFoundException($object, $id, $webspaceKey, $locale);
+        }
+
+        $webspace = clone $webspace;
+        $localization = $webspace->getLocalization($locale);
+
+        if (!$localization) {
+            throw new WebspaceLocalizationNotFoundException($object, $id, $webspaceKey, $locale);
+        }
+
+        $localization = clone $localization;
+        $localization->setXDefault(true);
+        $portal = new Portal();
+        $portal->setName($webspace->getName());
+        $portal->setKey($webspace->getKey());
+        $portal->setWebspace($webspace);
+        $portal->setXDefaultLocalization($localization);
+        $portal->setLocalizations([$localization]);
+        $portal->setDefaultLocalization($localization);
+        $webspace->setPortals([$portal]);
+
+        return new PortalInformation(RequestAnalyzer::MATCH_TYPE_FULL, $webspace, $portal, $localization);
```

Comment:
The question is if we should also pass an URL here... I can make something up, or we just leave it null (even if it is used in twig it should work, since it then output nothing...)